### DISCUSSION TITLE
How to use InputNumber for phone number

### DISCUSSION THREAD

**Discussion by ashikjs** [primefaces/community#111] [PrimeNG] [answered]
`
<p-inputNumber [(ngModel)]="value2" mode="decimal" inputId="withoutgrouping" [useGrouping]="false"> </p-inputNumber>
`

If we type here 01908471655 then it convert to 1908471655.
How we can use it for a phone number input .

**Comment by cetincakiroglu** [comment, marked as answer]
I suggest you use inputMask component for it, as you see we have a phone number example in the demo https://primeng.org/inputmask#mask

**Comment by ashikjs** [comment]
@cetincakiroglu  Thanks for your idea. 

But I already check inputMask. Some case for me like. 
Phone number and Telephone number both should able to input by one input box.

So user can input like below:
**Phone number:** 0190001115, 0192222255
**Telephone number:**  14056, 77789, 1736369, 11223333

so you can't give any validation except Number. So user can only input number.

**Reply by cetincakiroglu** [reply]
components that apply specific masks always return string value, because the mask cannot be applied to a number input. I guess you need to figure out how to convert the input you receive to something valid. Maybe regex would help you to validate.

The reason why inputnumber removes the 0 at the beginning, is that the component doesn't let you enter a number starting with 0. Maybe you can use the prefix input property of the inputnumber component for that?

**Reply by ashikjs** [reply]
NO prefix can not okay for me but I just using regex.

I except inputnumber should have opportunity to use for only number purpose. But i am agree with about remove zero from input if it is **mood decimal**

**Reply by cetincakiroglu** [reply]
Regex is more solid solution in your case
